plugins/nf-azure/src/main/nextflow/cloud/azure/file/AzPathFactory.groovy

```diff
@@ -48,6 +48,11 @@ class AzPathFactory extends FileSystemPathFactory {
         if( uri.startsWith('az:///') )
             throw new IllegalArgumentException("Invalid Azure path URI - make sure the schema prefix does not container more than two slash characters - offending value: $uri")
 
+        final storageAccountName = AzConfig.getConfig().storage().accountName
+        if (uri.contains(storageAccountName)){
```

Comment:
This sounds too fragile, what it that string is on the path ?

Reply:
Good point! should remove only from the beginning of the path
I'll update

Reply:
If you can isolate the URI hostname, should be able to split on the dot into storage account name and bucket name